exporter.go

```diff
@@ -422,3 +431,22 @@ func (e *Exporter) addOrReuseLayer(image imgutil.Image, layer layers.Layer, prev
 	e.Logger.Debugf("Layer '%s' SHA: %s\n", layer.ID, layer.Digest)
 	return layer.Digest, image.AddLayerWithDiffID(layer.TarPath, layer.Digest)
 }
+
+func (e *Exporter) makeBuildReport(layersDir string) (BuildReport, error) {
+	if e.PlatformAPI.Compare(api.MustParse("0.5")) < 0 { // platform API < 0.5
+		return BuildReport{}, nil
+	}
+	var out []BOMEntry
+	for _, bp := range e.Buildpacks {
+		if api.MustParse(bp.API).Compare(api.MustParse("0.5")) < 0 { // buildpack API < 0.5
+			continue
+		}
+		var bpBuildReport BuildReport
+		bpBuildTOML := filepath.Join(layersDir, bp.ID, "build.toml")
+		if _, err := toml.DecodeFile(bpBuildTOML, &bpBuildReport); err != nil && !os.IsNotExist(err) {
+			return BuildReport{}, err
+		}
+		out = append(out, withBuildpack(bp, bpBuildReport.BOM)...)
```

Comment:
I think we are accidentally allowing top-level versions here as well. I think we should either be validating that they aren't set or using a different struct types in different situations to avoid these gotchas.

Reply:
If a buildpack tries to write a top-level version to the BOM in either build.toml or launch.toml, is that a failure case? That would require a "validator" in order to detect (I believe).

At first I was leaning toward separate structs, but that would require some sort of duplication of both `Require` and `BOMEntry` (and converting of structs during `Build()`) which started to feel more confusing than just adding the logic at the time of reading the files.

Reply:
I lean towards erroring even though we don't do that for unrecognized keys just to ensure everyone migrate the version to the metadata section in this new API. At the very least a top-level version shouldn't end up in `report.toml`.